lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -397,4 +404,55 @@ void SetMessageManagerHandler::idle_mode_change_sessions_handle(
   return;
 }
 
+/* This function callled when specific IMSI service
+ * request is received from AMF.
+ */
+void SetMessageManagerHandler::service_handle_request_on_paging(
+    const SetSmNotificationContext& notif,
+    std::function<void(Status, SmContextVoid)> response_callback) {
+  // extract IMSI value from proto
+  auto imsi           = notif.common_context().sid().id();
+  auto session_map    = session_store_.read_sessions({imsi});
+  int count           = 0;
+  auto session_update = SessionStore::get_default_session_update(session_map);
+  for (auto& session : session_map[imsi]) {
+    if (session->get_state() == INACTIVE) {
+      auto session_id                        = session->get_session_id();
+      SessionStateUpdateCriteria& session_uc = session_update[imsi][session_id];
+      m5g_enforcer_->m5g_move_to_active_state(imsi, session, notif, session_uc);
+      bool update_success = session_store_.update_sessions(session_update);
+      if (!update_success) {
+        MLOG(MINFO) << "Operation aborted in  middle"
+                    << " session id: " << session->get_session_id() << imsi
+                    << " of imsi";
+        continue;
+      }
+      MLOG(MDEBUG) << "Successfully updated SessionStore "
+                   << " session id: " << session->get_session_id() << imsi
+                   << " of imsi";
+      count++;
+    }
+  }
+  if (!count) {
+    MLOG(MINFO) << " No Valid session found for IMSI: " << imsi;
+    Status status(grpc::NOT_FOUND, "Sesion Not found");
+    response_callback(status, SmContextVoid());
+    return;
+  }
+  bool update_success = session_store_.update_sessions(session_update);
+  if (!update_success) {
+    Status status(grpc::ABORTED, "update operation aborted in  middle");
+    response_callback(status, SmContextVoid());
+    return;
+  }
+  // Send  Paging service request response back to AMF
+  auto session = session_map[imsi].begin();

```

Comment:
Is the session guaranteed to be at the front? of the vector?

Reply:
Yes, for valid imsi/session front element of the SessionState vector contains session information.